From f774b65ccfcb34a015b9c6c2058e3717f7fcd14e Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lb@semihalf.com>
Date: Thu, 18 Oct 2018 10:46:34 +0300
Subject: [PATCH 0318/1051] octeontx: fix alloc failure when DMA_CMA is set

When DMA_CMA option was set in kernel configuration then allocation
for SSO and PKO failed because it tried to allocate much more memory
than really needed.

Signed-off-by: Lukasz Bartosik <lb@semihalf.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
index fc7ace8ad879..15d809ad8091 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
@@ -141,6 +141,9 @@ static int fpa_vf_addmemory(struct fpavf *fpa, u64 num_buffers, u32 buf_len)
 	u32 i, j, ret = 0;
 
 	chunk_size = MAX_ORDER_NR_PAGES * PAGE_SIZE;
+	if (chunk_size > num_buffers * buf_len)
+		chunk_size = num_buffers * buf_len;
+
 	buffs_per_chunk = chunk_size / buf_len;
 	fpa->vhpool_memvec_size = (num_buffers + buffs_per_chunk - 1) /
 				   buffs_per_chunk;
-- 
2.17.1

